psd-web/app/controllers/investigations/corrective_actions_controller.rb

```diff
@@ -1,7 +1,48 @@
 class Investigations::CorrectiveActionsController < ApplicationController
   def show
-    @investigation = Investigation.find_by!(pretty_id: params[:investigation_pretty_id])
+    @investigation = Investigation.find_by!(pretty_id: params[:investigation_pretty_id]).decorate
+    authorize @investigation, :view_non_protected_details?
+    @corrective_action = @investigation.corrective_actions.find(params[:id]).decorate
+  end
+
+  def edit
+    @investigation = Investigation.find_by!(pretty_id: params[:investigation_pretty_id]).decorate
     authorize @investigation, :view_non_protected_details?
```

Comment:
Please change: users need to have update permissions to be able to load this view.

```suggestion
    authorize @investigation, :update?
```